### PR TITLE
GCP multi-zone support

### PR DESCRIPTION
**Pull request by TheJokersThief**
* Adds support for autoscaling in multiple zones in one region on GCP.
* If a region is specified via `DRONE_GOOGLE_REGION`, it will randomly choose a zone within that region and ignore the zone specified in `DRONE_GOOGLE_ZONE`
* When destroying an instance, the destroy method will use the `Region` (which is actually a zone ID)

Quick question: I wasn't totally sure on the method for adding and recognising an optional environment variable, so please let me know if I missed something there :)

### CODE REVIEW

File: drivers/google/destroy.go
Comment (bradrydzewski) [inline drivers/google/destroy.go:19]: @TheJokersThief I made a minor change to preserve the existing behavior.  The google docs did not mention that region could be used, and I don't have enough experience with Google Cloud to confidently merge the pull request without spinning up a full environment and running some tests (which would be time consuming).  So in the interest of getting this merged quickly, I figured maybe we could implement the above logic as a compromise? Look ok to you?

**Comment by bradrydzewski** [comment]
lgtm pending answer to my question

**Review by bradrydzewski — APPROVED** [review]

File: drivers/google/create.go
Comment (bradrydzewski) [inline drivers/google/create.go:63]: you could probably simplify as `p.zone = path.Base(zoneSelfLink)`

Comment (bradrydzewski) [reply drivers/google/create.go:63]: sorry, one gotcha here is that `p.zone` is not going to be thread safe to write to.  do we need to overwrite `p.zone` or can this be stored in a local variable?

File: drivers/google/destroy.go
Comment (bradrydzewski) [reply drivers/google/destroy.go:19]: also does `waitZoneOperation` work with a region?

Comment (bradrydzewski) [reply drivers/google/destroy.go:19]: also just want to confirm that if we pass `region` to the `Delete` function, will this have any impact on the subsequent `waitZoneOperation`?  Sorry, I'm not too familiar with gcp.

File: drivers/google/create.go
Comment (TheJokersThief) [reply drivers/google/create.go]: You are right on both accounts - thanks :) This also uncovered that the operation wait function relied on having `p.zone` and succeeded during tests because I was obviously running that one test synchronously 😅 So I've changed that function to take a `zone` as an argument.

**Comment by bradrydzewski** [comment]
@TheJokersThief thanks, the code looks great.  I do want to ask a few design questions before we merge since there are a few different ways we could solve this problem.  Please note that these are just probing questions, I'm not advocating for or suggesting any changes, but want to make sure we consider a few different options before we finalize this.

1. do we want to make the API call to get the list of zones every time the install function is called, or is this something we would want to execute once and then cache?
2. if we want to make the API call every time, if the API call fails we currently error.  Instead of returning an error we could write the error message to the logs, but then fallback to the default zone.  This might make it slightly more resilient in the event the Google API is misbehaving (although not sure this is a problem in practice)
3. is it possible that someone would want to use a subset of zones in a region?  If yes, perhaps instead of providing a region and dynamically querying region for a list of zones, we allow the end-user to configure one or many zones instead of a single zone.

**Comment by TheJokersThief** [comment]
That's a really good point 🤔 Yep, I think having someone supply their list of zones (option 3) would be a good idea :) I can't think of a scenario where someone would want to use just a subset of zones, but it does feel like the option that would give users the most control.

Because `DRONE_GOOGLE_ZONE` is a variable, it would probably need to be a string of comma-separated values, right?

Also, should it be renamed `[...]_ZONES`? We could piggyback off this to support the singular name for backwards compatibility: https://github.com/drone/autoscaler/blob/master/config/load.go#L18-L21

**Comment by bradrydzewski** [comment]
> it would probably need to be a string of comma-separated values, right?

Yes, this would be automatically supported by the library we use to decode environment variable into the structure.  So we can change from a string to string slice.

```diff
-Zone                string            `envconfig:"DRONE_GOOGLE_ZONE"`
+Zone                []string          `envconfig:"DRONE_GOOGLE_ZONE"`
```

This would not have any adverse impact on existing users since a string value would be interpreted as a single value in a comma-separated array. Both of the below examples would decode into a string slice:

```text
DRONE_GOOGLE_ZONE=us-central1-a
DRONE_GOOGLE_ZONE=us-central1-a,us-central1-b
```

File: drivers/google/option.go
Comment (bradrydzewski) [inline drivers/google/option.go:137]: one minor change for consistency ... can you do `WithZones(zones ...string)` similar to this:
https://github.com/drone/autoscaler/blob/350f5a87a679c5797f9d35a1b448362843cdcf5f/drivers/google/option.go#L96

File: drivers/google/option.go
Comment (bradrydzewski) [inline drivers/google/option.go:144]: can we remove?

**Comment by bradrydzewski** [comment]
@TheJokersThief I've added two minor comments and then we should be good to merge.  Thanks for your patience through the review process!

File: drivers/google/option.go
Comment (TheJokersThief) [reply drivers/google/option.go:144]: Yep and removed :)

File: drivers/google/option.go
Comment (TheJokersThief) [reply drivers/google/option.go]: Got it, thanks 😄

**Comment by TheJokersThief** [comment]
@bradrydzewski No worries at all :) Thanks for working through it with me! I look forward to whenever this can be merged, it'll help Drone's GCP story quite a bit.

**Comment by bradrydzewski** [comment]
thanks!  I'll give it a final sanity check tomorrow and should have it merged this weekend.

**Review by bradrydzewski — APPROVED** [review]

**Comment by bradrydzewski** [comment]
@TheJokersThief this looks good and is ready to merge.  I may want to tag a release from master before I merge, just in case someone needs to revert back for some unknown reason (sorry, I am sometimes a bit overly cautious).  I have to write up the release notes which will take some time since we merged a ton of pull requests last week, but once complete, we can merge and cut a new minor release with your changes.

**Pull request merged**
